### PR TITLE
Add tests for Engine Cache

### PR DESCRIPTION
**Pull request by TimAbraldes**

### CODE REVIEW

File: tests/nodeUnit.js [hunk 1]
```diff
@@ -22,10 +22,60 @@ define(function(require) {
   const bdd = require('intern!bdd');
   const assert = require('intern/chai!assert');
   const fs = require('intern/dojo/node!fs');
+  const path = require('intern/dojo/node!path');
 
   const publicDir = 'dist/public';
 
+  function rm(filepath) {
```
Comment (marco-c) [inline tests/nodeUnit.js]: You can use `del` instead of rolling your own.

https://www.npmjs.com/package/del

File: tests/nodeUnit.js [hunk 2]
```diff
@@ -181,5 +223,87 @@ define(function(require) {
...
+
+    bdd.describe('Cache', function() {
+      const cache = require('intern/dojo/node!../../../../engine/cache').default;
+      const cacheDir = 'tests/support/var/engineCache';
+      const fetchMock = require('intern/dojo/node!fetch-mock');
+
+      bdd.before(function() {
+        const dir = path.dirname(cacheDir);
+
+        return new Promise(function(resolve, reject) {
```
Comment (marco-c) [inline tests/nodeUnit.js]: I'd use the sync version of the functions here, to make it more readable.

**Comment by marco-c** [comment]
Why do you want to make the engine cache async? It's just part of the build process, so does it really matter if it's async or sync?

**Comment by digitarald** [comment]
+1 for @marco-c's point  - can you make a case for async cache, @TimAbraldes ?

**Comment by TimAbraldes** [comment]
> Why do you want to make the engine cache more async? It's just part of the build process, so does it really matter if it's async or sync?

While reading code to figure out what tests to write, I just happened to notice that the engine cache's API looked async but the actual calls were sync. I haven't had issues with the build being slow (and I doubt that this would make it much faster) so I don't have a strong case for it.

I'll remove the bits of this patch that change the cache and just keep the tests.

**Comment by TimAbraldes** [comment]
I'm not sure what's up with that build failure; it definitely doesn't seem related to this patch

**Comment by marco-c** [comment]
Looks like Travis failed to download Node.js v4 and used 0.12 instead (where Promise isn't defined).

**Pull request merged**